src/osd/OSD.cc

```diff
@@ -7062,6 +7050,44 @@ void OSD::sched_scrub()
 
 
 
+vector<OSDHealthMetric> OSD::get_health_metrics()
+{
+  vector<OSDHealthMetric> metrics;
+  {
+    utime_t oldest_secs;
+    const utime_t now = ceph_clock_now();
+    auto too_old = now;
+    too_old -= cct->_conf->get_val<double>("osd_op_complaint_time");
+    int slow = 0;
+    auto count_slow_ops = [&](TrackedOp* op) {
+      if (op->get_initiated() < too_old) {
+	slow++;
```

Comment:
if this change looks sane, i will

1. add the exponential backoff back and 
1. store up to `osd_op_log_threshold` slow requests on OSD in memory. 
1. add a command to dump these slow requests.